drivers/hwinfo/hwinfo_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2023 Piotr Dymacz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <driverlib/sys_ctrl.h>
+#include <inc/hw_ccfg.h>
+#include <inc/hw_fcfg1.h>
+#include <inc/hw_memmap.h>
+
+#ifdef CONFIG_HWINFO_CC13XX_CC26XX_USE_BLE_MAC
+#define CC13XX_CC26XX_DEVID_SIZE	6
+#else
+#define CC13XX_CC26XX_DEVID_SIZE	8
+#endif
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint8_t *mac;
+
+#ifdef CONFIG_HWINFO_CC13XX_CC26XX_USE_BLE_MAC
+#ifdef CONFIG_HWINFO_CC13XX_CC26XX_USE_FACTORY
+	mac = (uint8_t *)(FCFG1_BASE + FCFG1_O_MAC_BLE_0);
+#else
+	if (sys_read32(CCFG_BASE + CCFG_O_IEEE_BLE_0) != 0xFFFFFFFF &&
+	    sys_read32(CCFG_BASE + CCFG_O_IEEE_BLE_1) != 0xFFFFFFFF) {
+		mac = (uint8_t *)(CCFG_BASE + CCFG_O_IEEE_BLE_0);
+	} else {
+		mac = (uint8_t *)(FCFG1_BASE + FCFG1_O_MAC_BLE_0);
+	}
+#endif
```

Comment:
Can you add comment on which #if #else path is closing?
Otherwise looks good to me.
Thanks!

Reply:
Done.
Was wondering whether indentations of preprocessor statements are accepted/allowed?
E.g. something like:

```
#ifdef ...
#  ifdef ...
#  else
#  endif
#endif
```

Reply:
Just the comments as far as I know.